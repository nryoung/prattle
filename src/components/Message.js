import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import * as colors from '../styles/colors';

const StyledMessage = styled.div`
  flex: 1 1 100%;
  margin: 0 0 16px 0;
`;

const MessageUser = styled.div`
  margin-bottom: 8px;
`;

const MessageTextWrapper = styled.div`
  display: inline-block;
  border-radius: 8px;
  word-break: break-word;
  background-color: ${colors.acidWash};
  padding: 8px;
`;

const MessageText = styled.span`
  color: ${colors.pureWhite};
`;

const Message = ({ message }) => (
  <StyledMessage>
    <MessageUser>{message.userId}:</MessageUser>
    <MessageTextWrapper>
      <MessageText>{message.text}</MessageText>
    </MessageTextWrapper>
  </StyledMessage>
);

export default Message;
