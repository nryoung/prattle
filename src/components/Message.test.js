import React from 'react';
import renderer from 'react-test-renderer';
import Message from './Message';

const props = {
  message: {
    sender: { name: ':senderName' },
    text: ':text',
  },
};

it('it renders', () => {
  const component = renderer.create(<Message {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
