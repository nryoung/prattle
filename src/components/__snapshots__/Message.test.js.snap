// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="sc-bdVaJa fvrKfw"
>
  <div
    className="sc-bwzfXH jRpyJY"
  >
    :senderName
    :
  </div>
  <div
    className="sc-htpNat kdcAzZ"
  >
    <span
      className="sc-bxivhb Igthg"
    >
      :text
    </span>
  </div>
</div>
`;
